radix-engine-tests/tests/balance_changes.rs

```diff
@@ -220,12 +220,12 @@ fn test_balance_changes_when_transferring_non_fungibles() {
             account.into() => indexmap!(
                 resource_address => BalanceChange::NonFungible {
                     added: BTreeSet::new(),
-                    removed: btreeset!(NonFungibleLocalId::integer(2))
+                    removed: btreeset!(NonFungibleLocalId::integer(3))
```

Comment:
I am quite surprised by this change o_O

(was it definitely wrong before? or is maybe the current impl wrong?)

Reply:
It looks like somehow the ordering of the keys in the vault has changed - which suggests that the DbMapper for sorted keys has changed? Which I guess it has?

Reply:
It's because the iteration over the index has changed due to the new sort key. Or do you think this should have remained the same?

Reply:
if the "index" you refer to is _not_ a sorted index, then yeah, totally - we changed the DB key serialization (hash prefix, discriminator, whatever), so the iteration order might have changed 👍 

AFAIU, this test should not care about which exact NFLI was transferred, right? (it just should assert that account A lost it and account B got it). I adjusted these asserts.